endpoints-framework/src/main/java/com/google/api/server/spi/discovery/DiscoveryGenerator.java

```diff
@@ -174,13 +193,21 @@ private RestDescription writeApi(ApiKey apiKey, Iterable<ApiConfig> apiConfigs,
       if (config.getCanonicalName() != null) {
         doc.setCanonicalName(config.getCanonicalName());
       }
+      allScopes.addAll(AuthScopeExpressions.encode(config.getScopeExpression()));
       for (ApiMethodConfig methodConfig : config.getApiClassConfig().getMethods().values()) {
         if (!methodConfig.isIgnored()) {
-          writeApiMethod(config, servicePath, doc, methodConfig, repo);
+          writeApiMethod(config, servicePath, doc, methodConfig, repo, allScopes);
         }
       }
     }
 
+    LinkedHashMap<String, ScopesElement> scopeElements = new LinkedHashMap<>();
+    for (String scope : allScopes) {
```

Comment:
Presumably this doesn't work when multiple scopes are required? What would happen if the scope expression is "email profile" for example?

Reply:
Indeed, but is this how the @Api*.scopes field is supposed to be used? 
It is a String array, so I assumed each scope must be declared independently.
i.e. @ApiMethod(scopes = {"email", "profile"}) rather than @ApiMethod(scopes = {"email profile"})

Reply:
This probably needs to be documented better, but the comma separation is a logical OR for scopes. To require multiple scopes simultaneously you are supposed to use one string with space separation.